### PR TITLE
M01 - assessment review

### PR DESCRIPTION
**Pull request by ahcarter22**
Submitting assessment with corrections to the display / "fringe" cases. Method now returns only the 5 capsules above and below the input- within the range.

### CODE REVIEW

**Comment by ahcarter22** [comment]
Also, I must have worked on array exercises within this branch- apologies.

File: week-01/assessment/capsule-hotel/Capsule.java [hunk 1]
```diff
@@ -0,0 +1,234 @@
+
+import java.util.Scanner;
+
+public class Capsule {
+    public static void main(String[] args) {
+        //  initialization method that will immediately get called, responsible for user input of capsule #
+        appInitialization();
+    }
+
+    public static void appInitialization() {
```
Comment (dexterfitch) [inline week-01/assessment/capsule-hotel/Capsule.java:10]: Nice abstraction from main() to create an initializer method, good thinking

File: week-01/assessment/capsule-hotel/Capsule.java [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+        // prompt to welcome user and ask for capsule number
+        System.out.println("---------------------------");
+        System.out.println("Welcome to Capsule-Capsule!");
+        System.out.println("---------------------------");
+        System.out.print("Enter the number of capsules available: ");
+        // Scanner to take in capsule number
```
Comment (dexterfitch) [inline week-01/assessment/capsule-hotel/Capsule.java:16]: Great use of comments, love to see it

File: week-01/assessment/capsule-hotel/Capsule.java [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+
+        System.out.println("Guest Check In");
+        System.out.println("---------------");
+        System.out.print("Please enter guest's name: ");
+
+        Scanner console = new Scanner(System.in);
+        String input = console.nextLine();
+        // variable to use and exit while loop
+        boolean isGuest = false;
+        // cannot be an empty string, but calls check in method again
```
Comment (dexterfitch) [inline week-01/assessment/capsule-hotel/Capsule.java:92]: Nice input validations throughout, well done!

File: week-01/assessment/capsule-hotel/Capsule.java [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+    public static void checkIn(String[] arr) {
+
+        System.out.println("Guest Check In");
+        System.out.println("---------------");
+        System.out.print("Please enter guest's name: ");
+
+        Scanner console = new Scanner(System.in);
+        String input = console.nextLine();
+        // variable to use and exit while loop
+        boolean isGuest = false;
```
Comment (dexterfitch) [inline week-01/assessment/capsule-hotel/Capsule.java:91]: Nice intuitive naming schema for variables throughout

File: week-01/assessment/capsule-hotel/Capsule.java [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+    // method for viewing the guests
+    public static void displayGuest(String[] arr){
+        System.out.println("View Guests");
+        System.out.println("-----------");
+        System.out.print("Please enter a capsule number: ");
+        Scanner console = new Scanner(System.in);
+        // declare capsuleNum variable
+        int capsuleNum;
+        // stays in the while loop, while the input is entered
+        boolean isGuest = false;
```
Comment (dexterfitch) [inline week-01/assessment/capsule-hotel/Capsule.java:180]: This is the only variable name I might consider changing - this is less "isGuest = false" and more like "isDonePrintingRooms = false" or something

**Review by dexterfitch — APPROVED** [review]
Comments marked **NOT ESSENTIAL** - just a tip, no big!
Comments marked **ESSENTIAL** - must implement changes 

YOU HAVE NO **ESSENTIAL** CHANGES! Great work!

### Requirements Checklist - MEETS REQUIREMENTS!

* [x] Features (each feature complete, works without errors)
  * [x] Check In
  * [x] Check Out
  * [x] View Guests
* [x] Methods (makes each high level scenario into a separate method)
* [x] Array Management
  * [x] View Guests displays properly even at the ends of the array
  * [x] Guests are added and removed from the array
* [x] Decisions and Repetition (uses appropriate decision and repetition statements)
* [x] Naming (uses consistent and meaningful names for variables and methods)
* [x] Code Format (Java code formatting is idiomatic)
* [x] Data Types (no unused variables; data types are sensible)

**Pull request closed without merging**